api/filter_dimension_options.go

```diff
@@ -54,10 +56,19 @@ func (api *FilterAPI) getFilterBlueprintDimensionOptions(w http.ResponseWriter,
 	bytes, err := json.Marshal(options)
 	if err != nil {
 		log.ErrorC("failed to marshal filter blueprint dimension options into bytes", err, logData)
+		if auditErr := api.auditor.Record(r.Context(), getOptionsAction, actionUnsuccessful, auditParams); auditErr != nil {
```

Comment:
This is just an opinion BUT - if you changed the return of your function to `([]byte, error)` then you could return the marshalled bytes and save yourself this extra audit unsuccessful step.